protocol-units/execution/fin-view/src/fin_view.rs

```diff
@@ -0,0 +1,200 @@
+				.unwrap(),
+		)
+	}
+
+	#[tokio::test]
+	async fn test_set_finalized_block_height_get_api() -> Result<(), anyhow::Error> {
+		// Create an Executor and a FinalityView instance from the environment configuration.
+		let config = AptosConfig::try_from_env()?;
+		let executor = Executor::try_from_config(&config)?;
+		let finality_view = FinalityView::try_from_config(
+			executor.db.reader.clone(),
+			executor.mempool_client_sender.clone(),
+			&config,
+		)?;
+
+		// Initialize a root account using a predefined keypair and the test root address.
+		let root_account = LocalAccount::new(
+			aptos_test_root_address(),
+			AccountKey::from_private_key(config.private_key.clone()),
+			0,
+		);
+
+		// Seed for random number generator, used here to generate predictable results in a test environment.
+		let seed = [3u8; 32];
+		let mut rng = ::rand::rngs::StdRng::from_seed(seed);
+
+		// Create a transaction factory with the chain ID of the executor.
+		let tx_factory = TransactionFactory::new(config.chain_id.clone());
+
+		let mut account_addrs = Vec::new();
+
+		// Simulate the execution of multiple blocks.
+		for _ in 0..3 {
+			let (epoch, round) = executor.get_next_epoch_and_round().await?;
+
+			let block_id = HashValue::random(); // Generate a random block ID for each block.
+
+			// Clone the signer from the executor for signing the metadata.
+			let signer = executor.signer.clone();
+			// Get the current time in microseconds for the block timestamp.
+			let current_time_micros = chrono::Utc::now().timestamp_micros() as u64;
+
+			// Create a block metadata transaction.
+			let block_metadata = Transaction::BlockMetadata(BlockMetadata::new(
+				block_id,
+				epoch,
+				round,
+				signer.author(),
+				vec![],
+				vec![],
+				current_time_micros,
+			));
+
+			// Generate new accounts and create transactions for each block.
+			let mut transactions = Vec::new();
+			transactions.push(block_metadata.clone());
+
+			// Each block will contain a transaction creating an account.
+			let new_account = LocalAccount::generate(&mut rng);
+			account_addrs.push(new_account.address());
+
+			let user_account_creation_tx = root_account.sign_with_transaction_builder(
+				tx_factory.create_user_account(new_account.public_key()),
+			);
+			transactions.push(Transaction::UserTransaction(user_account_creation_tx));
+
+			// Group all transactions into an unsharded block for execution.
+			let executable_transactions = ExecutableTransactions::Unsharded(
+				transactions.into_iter().map(SignatureVerifiedTransaction::Valid).collect(),
+			);
+			let block = ExecutableBlock::new(block_id.clone(), executable_transactions);
+			executor.execute_block(block).await?;
+		}
+
+		finality_view.set_finalized_block_height(2)?;
+
+		// Retrieve the executor's API interface and fetch the accounts
+		let apis = finality_view.get_apis();
+		let context = apis.accounts.context.clone();
```

Comment:
This is using context, maybe let's do the same thing using `apis.accounts` itself for completeness--in case there's something the API does that we weren't expecting. That or let's link to the Aptos core implementation and leave a comment here that explains why such a test is unccessary.

Reply:
The APIs in aptos-api expose methods that use path arguments and other scaffolding from the poem web server, so I called context directly to not have to deal with this.
The only exception I saw is `TransactionsApi::get_transaction_by_hash_inner`, perhaps defined as an escape hatch for similar testing purposes, and we use it in executor tests.
But you are right, I'll look into how to make it as close as possible to real life.

Reply:
Sweet! As I said, the comment would suffice too.

Reply:
aptos-api methods bound for OpenAPI are not public, so I exposed one and used it in https://github.com/movementlabsxyz/movement/pull/91/commits/f25dc248913593378a094acc62ecba74be4245d0